src/ServerlessPluginBespoken.ts

```diff
@@ -0,0 +1,407 @@
+      if (!serverlessFunctions[key].handler) {
+        continue;
+      }
+      sFunction = serverlessFunctions[key];
+      break;
+    }
+  }
+
+  if (!sFunction) {
+    throw new Error("No available function found");
+  }
+
+  return createFunctionObject(sFunction);
+};
+
+const createFunctionObject = (serverlessFunction: { handler: string }) => {
+  const splitHandler = serverlessFunction.handler.split(".");
+  const file = splitHandler[0];
+  const exportedFunction = splitHandler[1];
+  return {
+    file,
+    exportedFunction
+  };
+};
+
+/**
+ * Defines command:
+ * - `serverless proxy`
+ *    Initiates connection to public bespoken proxy so requests to public url are forwarded to local machine and starts 
+ *    a local server that responds to requests by dispatching the url path to the appropriate lambda function.
+ * 
+ * - `serverless deploy --enablePassThru`
+ *   When --enablePassThru is passed on command line, the plugin will intercept the serverless package lifecycle and replace the deployed bundle with passthru lambda handlers
+ *   that forward requests to the bespoken proxy server (and subsequently back to developer machine)
+ */
+export class ServerlessPluginBespoken {
+  private serverless: any;
+  private options: any;
+  private proxy: BSTProxy;
+  private originalServicePath: string;
+  private passThruServicePath: string;
+
+  private get selectedFunctionFromCommandLine(): string | undefined {
+    return this.options.function;
+  }
+
+  private get functionsFromServerlessConfig(): any {
+    return this.serverless.service.functions;
+  }
+
+  private get environmentVariablesFromServerlessConfig(): object {
+    const env = this.serverless.service.provider.environment;
+    if (env) {
+      this.serverless.cli.log(
+        "Configuring environment variables from serverless config"
+      );
+    } else {
+      this.serverless.cli.log(
+        "No Environment variables found in serverless config"
+      );
+    }
+    return env || {};
+  }
+
+  private get handlers(): string[] {
+    const handlers = this.serverless.service.functions || {};
+
+    return Object.entries(handlers).map(([_, lambdaDefinition]) => {
+      return lambdaDefinition.handler;
+    });
+  }
+
+  private get enableSecurity(): boolean {
+    return !(this.options.secure == null);
+  }
+
+  public commands = {
+    proxy: {
+      usage: "Plugin to call bespoken bst lambda service",
+      lifecycleEvents: ["start"],
+      options: {
+        function: {
+          usage:
+            "Specify the function in your config that you want to use" +
+            '(e.g. "--function myFunction" or "-f myFunction")',
+          required: false,
+          shortcut: "f"
+        },
+        secure: {
+          usage:
+            "Make proxy server require that 'secure' token be specified -- used to reduce likelihood of arbitrary hosts contacting your service via the proxy server",
+          required: false,
+          shortcut: "s"
+        }
```

Comment:
add a new option here for passthrough, passthrough should be selected as an option to enable by default  getting the function from the serverless config